### ISSUE TITLE
ECMWF UK 2017 - 2019 (Current steps and params)

### ISSUE THREAD

**Issue by peterdudfield** [openclimatefix/dagster-dags#56]
## 29-01-2024

Backfill launched to cover period 2018-2019 with priority 10: [view backfill in dagster](http://localhost:3003/overview/backfills/ohfhrpct)

Backfill Launched to cover period 2017-2018 with priority 10 (will run after the above): [view backfill in dagster](http://localhost:3003/overview/backfills/nnjepvfl)

## 30-01-2024

Issue with the ECMWF UK Backfill for 2017-2019: 

We are requesting from Mars the following: 
2 Meter Temperature,
10 Meter U component of wind,
10 Meter V Component of wind,
Direct Solar Radiation,
Downward UV Radiation at surface,
High Cloud Cover,
Medium cloud cover,
Low cloud cover,
Surface shortwave radiation downward,
Surface longwave radiation downward,
Total precipitation rate*,
Snow depth,
100 Meter U componenet of wind,
100 Meter V component of wind,
200 Meter U component of wind*,
200 Meter V component of wind*

But the ones marked with stars don't exist for that time period. 
200 Meter U and V component are only available from 2019-06-12
Total Precipitation Rate is only available from 2018-06-06

So currently the backfills are failing as MARS has a guard in place for downloading fewer fields than expected. I can put a safeguard in pace so these aren't being requested in older backfills but then those fields will be missing from the data.

Blocked by https://github.com/openclimatefix/nwp-consumer/issues/98

UPDATE 18:30
A fix has been implemented in the consumer. The backfills have been recreated.

## 01-02-2024

Downloads seem to be going very slowly. It might be a MARS queue issue, in which case there's nothing we can do. However, some seen to be getting stuck at the reading the of the file returned by MARS' "list" request. As such I've implemented a slightly more robust reading pattern as well as a debug log to enable double checking the problem. It could also just be that Leonardo's disk is full.

See the relevant PR here: https://github.com/openclimatefix/nwp-consumer/pull/101

## 02-02-2024

Backfill has ceased as /mnt/storage_b is full.
Recreated backfill in evening.

## 05-02-2024

Many errors in backfill due to "too many queued requests" from ECMWF. I have checked our web API activity and this morning it doesn't look like anything is queued, so I'm kicking it back off and hoping that it won't hit that cap again.

## 07-02-2024

500 partitions to go, at approximately 22 minutes per partition.

## 12-02-2024

Fewer than 100 to go.

## 13-02-2024

20 Left.

## 14-02-2024

Complete, minus a couple which have fallen victim to the missing fields error we saw earlier:
https://github.com/openclimatefix/nwp-consumer/issues/98

New issue for this tracked in https://github.com/openclimatefix/nwp-consumer/issues/112
